api/helpers/add-game-to-match.js

```diff
@@ -35,13 +35,13 @@ module.exports = {
       let numPlayer2Wins = 0;
 
       for (const priorGame of [game, ...relevantMatch.games]) {
-        if (priorGame.result === 0) {
+        if (priorGame.winner === priorGame.p0) {
           if (priorGame.p0 === relevantMatch.player1) {
             numPlayer1Wins++;
           } else {
             numPlayer2Wins++;
           }
-        } else if (priorGame.result === 1) {
+        } else if (priorGame.winner === priorGame.p1) {
           if (priorGame.p1 === relevantMatch.player1) {
             numPlayer1Wins++;
           } else {
```

Comment:
Now that game.winner replaces game.result I think we can significantly refactor this whole block. The wonky conditional logic around p0 and p1 was all about cross referencing which player was which and who won by comparing the result (indicating whether p0 or p1 won) against which player was p0 or p1 in the game, against which player is p1 vs p2 in the match. 

This can be much much simpler now. The match wants to know how many games each player has one. So just loop through the games and compare the winner attribute to the match's player1 and player2

Reply:
Writing the logic here the first time is what inspired me to write this issue up because the use of result (as opposed to the winner's id) made this so convoluted that I wanted to rework the database to clean up this and related logic